packages/Heading/src/Heading.styles.js

```diff
@@ -52,17 +40,28 @@ const underline = level => {
   return result;
 };
 
-export const dividerStyles = `
+export const DividerWrapper = styled.span`
   border-bottom: 2px solid ${tokens.color.blackLighten70};
   flex-grow: 1;
   margin-left: ${tokens.space};
 `;
 
-export const headingStyles = props => `
-  ${defaultHeadingStyles}
-  ${heading[`HEADING_LEVEL_${props.safeDisplayLevel || props.safeLevel}`]}
-  ${props.isHidden ? stylers.visuallyHidden : ""}
-  ${props.hasUnderline ? underline(props.safeDisplayLevel || props.safeLevel) : ""}
-  ${props.hasDivider ? "display: flex;" : ""}
-  ${props.isLight ? "font-weight: 400;" : ""}
+export const HeadingWrapper = styled.div`
+  align-items: center;
+  color: inherit;
+  font-weight: 700;
+  letter-spacing: 0;
+  ${stylers.lineHeight(-1)};
+  margin: ${stylers.spacer(2)} 0;
+
+  &:focus {
+    outline: none;
+  }
+  ${props => `
```

Comment:
Heres' an example fo a block with a bunch of props being used where it totally makes sense to wrap the whole block once in a function, instead of for each line.  And could make it more readable _not_ to destructure props at the beginning, since you'd have a list of 4 or 5.  But I still am not crazy about the way this approach scales down... 🤔